test/functional/discovered_hosts_controller_test.rb

```diff
@@ -294,6 +294,9 @@ def test_no_dns_rebuild_if_dns_pending
     hostgroup = prepare_hostgroup_for_dns_rebuild(host)
     Nic::Managed.any_instance.expects(:rebuild_dns).never
     Host::Managed.any_instance.stubs(:skip_orchestration?).returns(false)
+    Host::Managed.any_instance.stubs(:queue_puppetca_create).returns(true)
+    Host::Managed.any_instance.stubs(:queue_puppetca_certname_reset).returns(true)
+    Host::Managed.any_instance.stubs(:queue_puppetca_autosign_destroy).returns(true)
```

Comment:
There are two commits, the first one just reflect change in core: https://github.com/theforeman/foreman/pull/5580